### DISCUSSION TITLE
Adding support for controlling WebSockets devices

### DISCUSSION THREAD

**Discussion by ademuri** [artisan-roaster-scope/artisan#997] [Ideas]
I'm working on a roaster control board which connects to WiFi (using an ESP32). I have added support for the Artisan WebSockets interface. I would like to add support to Artisan for setting the temperature. If I write that, is there a possibility of merging it? Do you foresee any issues?

**Comment by poundy** [comment]
I'm not going to comment on the technical capability of websockets to be used as write-devices in the existing codebase, but I wonder if you've considered implementing modbus-TCP, a much more standard industrial control layer?

**Comment by ademuri** [comment]
I didn't realize that modbus-TCP existed - I'll look into that, thank you!

**Comment by MAKOMO** [comment]
I don't understand exactly what you mean by "setting the temperature", maybe you can expand on that a bit. Note that there are Web Socket command actions to write any JSON to the connected device, similar to MODBUS Commands. Thus you might be able to do what you intend by configuration instead of implementation.

**Reply by ademuri** [reply]
Ohh, I didn't understand how that worked - my controller is PID-capable, so I mean to change the setpoint for the PID controller. The configuration for WebSockets works well for this. Thanks!

**Comment by sakunamary** [comment]
> I'm working on a roaster control board which connects to WiFi (using an ESP32). I have added support for the Artisan WebSockets interface. I would like to add support to Artisan for setting the temperature. If I write that, is there a possibility of merging it? Do you foresee any issues?

For Your Information :
https://github.com/artisan-roaster-scope/artisan/discussions/866

You may make  your own protocol which control board talk to drummer with , WebSocket just a kind of tunnel to let the commands runs on.
